native/src/init/mount.cpp

```diff
@@ -70,11 +72,17 @@ static struct {
 } blk_info;
 
 static dev_t setup_block() {
+    const auto partition_map = load_partition_map();
     if (dev_list.empty())
         collect_devices();
 
     for (int tries = 0; tries < 3; ++tries) {
         for (auto &dev : dev_list) {
+            // use androidboot.partition_map as partname fallback.
+            if (dev.partname[0] == '\0') {
+                auto partname = get_partition_name_for_device(partition_map, dev.devname);
```

Comment:
can also inline `get_partition_name_for_device` with `find_if`

Reply:
I can't write it in a clean way with find_if.